### PR TITLE
[KeyVault] Removed redundant Microsoft Graph API calls for access policy in `Get-AzKeyVault`

### PR DESCRIPTION
**Pull request by BethanyZhou**
## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/KeyVault/KeyVault/Models/ModelExtensions.cs [hunk 1]
```diff
@@ -159,5 +171,80 @@ public static string GetDisplayNameForTenant(Guid id, IMicrosoftGraphClient grap
                 return string.Empty;
             return id.ToString();
         }
+
+        private static  List<PSADObject> GetObjectsByObjectIds(List<string> objectIds, IMicrosoftGraphClient graphClient)
```
Comment (BethanyZhou) [inline src/KeyVault/KeyVault/Models/ModelExtensions.cs:175]:
```suggestion
        private static List<PSADObject> GetObjectsByObjectIds(List<string> objectIds, IMicrosoftGraphClient graphClient)
```

File: src/KeyVault/KeyVault/ChangeLog.md [hunk 2]
```diff
@@ -18,6 +18,7 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Reduced number of MSGraph API calls for access policy.
```
Comment (isra-fel) [inline src/KeyVault/KeyVault/ChangeLog.md:21]:
```suggestion
* Removed redundant Microsoft Graph API calls for access policy cmdlets.
```

File: src/KeyVault/KeyVault/Models/PSKeyVaultAccessPolicy.cs [hunk 3]
```diff
@@ -46,6 +46,19 @@ public PSKeyVaultAccessPolicy(KeyVaultManagement.Models.AccessPolicyEntry s, IMi
             PermissionsToStorage = s.Permissions.Storage == null ? new List<string>() : new List<string>(s.Permissions.Storage);
         }
 
+        public PSKeyVaultAccessPolicy(KeyVaultManagement.Models.AccessPolicyEntry s, string displayName)
```
Comment (isra-fel) [inline src/KeyVault/KeyVault/Models/PSKeyVaultAccessPolicy.cs:49]: Constructor on line 36 can be written as 
```csharp
public PSKeyVaultAccessPolicy(KeyVaultManagement.Models.AccessPolicyEntry s, IMicrosoftGraphClient graphClient): this(s, ModelExtensions.GetDisplayNameForADObject(s.ObjectId, graphClient)) {}
```

File: src/KeyVault/KeyVault/Models/ModelExtensions.cs [hunk 4]
```diff
@@ -159,5 +171,91 @@ public static string GetDisplayNameForTenant(Guid id, IMicrosoftGraphClient grap
                 return string.Empty;
             return id.ToString();
         }
+
+        private static List<PSADObject> GetObjectsByObjectIds(List<string> objectIds, IMicrosoftGraphClient graphClient)
+        {
+            // todo: do we want to use 1000 as batch count in msgraph API?
```
Comment (isra-fel) [inline src/KeyVault/KeyVault/Models/ModelExtensions.cs:177]: @VeryEarly any thoughts?

**Review by isra-fel — DISMISSED** [review]

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**